cf-atcp-contact/src/main/java/org/broadinstitute/ddp/cf/ATCPStudyContact.java

```diff
@@ -0,0 +1,172 @@
+package org.broadinstitute.ddp.cf;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+import com.aventrix.jnanoid.jnanoid.NanoIdUtils;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.functions.HttpFunction;
+import com.google.cloud.functions.HttpRequest;
+import com.google.cloud.functions.HttpResponse;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.HttpMethod;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.broadinstitute.ddp.cf.json.FileUploadUrl;
+import org.broadinstitute.ddp.cf.json.StudyContactRequest;
+
+public class ATCPStudyContact implements HttpFunction {
```

Comment:
@YuriyPavlovBI I have refactored the implementation here, please take a close look. Some things I did:

* Refactored the recaptcha code out to a separate class to keep things clean. The `RecaptchaClient` is derived from work that @MocanaAtBroad had done in Pepper.
* Added more logging.
* Added check for `POST` HTTP method.
* Threw exceptions in appropriate places so they get picked up by GCP Error Reporting.
* Improved on request payload checking.

Reply:
LGTM